src/crimson/osd/osd.cc

```diff
@@ -391,12 +403,49 @@ seastar::future<> OSD::_send_alive()
   }
 }
 
+/*
+  Note re process:
+  as stop_asok_admin() as executed as part of a destruction process, we
+  take nothing for granted. All interfaces are checked for existence.
```

Comment:
This isn't actually part of the destructor.  The non-determinism is because it's called during at_exit, which might happen prior to start.  It's probably worth noting that.

Reply:
I modified the comment. Note that I did not mean the dtor, hence the general 'destruction'.